examples/end2end_tree_reconstruction_test.py

```diff
@@ -0,0 +1,356 @@
+#!/usr/bin/env python3
+
+import argparse
+from itertools import product
+import os
+import subprocess
+import sys
+import typing
+
+import alifedata_phyloinformatics_convert as apc
+from colorclade import draw_colorclade_tree
+import matplotlib.pyplot as plt
+import pandas as pd
+from teeplot import teeplot as tp
+
+from hstrat._auxiliary_lib import (
+    alifestd_calc_triplet_distance_asexual,
+    alifestd_collapse_unifurcations,
+    alifestd_count_leaf_nodes,
+    alifestd_mark_node_depth_asexual,
+    alifestd_prune_extinct_lineages_asexual,
+    alifestd_try_add_ancestor_list_col,
+)
+
+ReconstructionResult = typing.Dict[
+    typing.Union[
```

Comment:
I'd just do `str` to `DataFrame` as it's more likely that we'll want to change or add over time (compared to a function sig)